src/js/ripple/remote.js

```diff
@@ -9,117 +10,64 @@
+var Request = require('./request').Request;
+var Server = require('./server').Server;
+var Amount = require('./amount').Amount;
+var Currency = require('./currency').Currency;
+var UInt160 = require('./uint160').UInt160;
+var UInt256 = require('./uint256').UInt256;
+var Transaction = require('./transaction').Transaction;
+var Account = require('./account').Account;
+var Meta = require('./meta').Meta;
+var OrderBook = require('./orderbook').OrderBook;
+var PathFind = require('./pathfind').PathFind;
 var SerializedObject = require('./serializedobject').SerializedObject;
-var RippleError      = require('./rippleerror').RippleError;
-var utils            = require('./utils');
-var sjcl             = require('./utils').sjcl;
-var hashprefixes     = require('./hashprefixes');
-var config           = require('./config');
-var log              = require('./log').internal.sub('remote');
-
-/**
- *    Interface to manage the connection to a Ripple server.
- *
- *    This implementation uses WebSockets.
- *
- *    Keys for opts:
- *
- *      trace
- *      max_listeners      : Set maxListeners for remote; prevents EventEmitter warnings
- *      connection_offset  : Connect to remote servers on supplied interval (in seconds)
- *      trusted            : truthy, if remote is trusted
- *      max_fee            : Maximum acceptable transaction fee
- *      fee_cushion        : Extra fee multiplier to account for async fee changes.
- *      servers            : Array of server objects with the following form
- *      canonical_signing  : Signatures should be canonicalized and the "canonical" flag set
- *
- *         {
- *              host:    <string>
- *            , port:    <number>
- *            , secure:  <boolean>
- *         }
- *
- *    Events:
- *      'connect'
- *      'disconnect'
- *      'state':
- *      - 'online'        : Connected and subscribed.
- *      - 'offline'       : Not subscribed or not connected.
- *      'subscribed'      : This indicates stand-alone is available.
- *
- *    Server events:
- *      'ledger_closed'   : A good indicate of ready to serve.
- *      'transaction'     : Transactions we receive based on current subscriptions.
- *      'transaction_all' : Listening triggers a subscribe to all transactions
- *                          globally in the network.
- *
- *    @param opts      Connection options.
- *    @param trace
- */
-
-function Remote(opts, trace) {
+var RippleError = require('./rippleerror').RippleError;
+var utils = require('./utils');
+var sjcl = require('./utils').sjcl;
+var hashprefixes = require('./hashprefixes');
+var config = require('./config');
+var log = require('./log').internal.sub('remote');
+
+/**
+ * Interface to manage connections to rippled servers
+ *
+ * @param {Object} Connection options.
+ */
+
+function Remote(opts) {
   EventEmitter.call(this);
 
   var self = this;
-  var opts = opts || { };
+  var opts = lodash.extend(this, Remote.DEFAULTS, opts);
```

Comment:
Nope, we want to extend `this`. Classes such as `TransactionManager` access configuration from their `remote` reference. This is the way it currently works. We could add an options container object like you suggest, and I'd be weakly in favor of it